### ISSUE TITLE
How to get API object using access_token only

### ISSUE THREAD

**Issue by keponk** [halcy/Mastodon.py#131]
Hi, when working with a similar [API in nodejs](https://github.com/jhayley/node-mastodon), I noticed I only needed the access token to work with a Mastodon instance. 
In contrast with this API it seems I'm forced to provide a client_id (and secret).
Example:
```
var M = new Masto({
  access_token: '...',
  timeout_ms: 60*1000,  // optional HTTP request timeout to apply to all requests.
  api_url: 'https://gay.crime.team/api/v1/', // optional, defaults to https://mastodon.social/api/v1/
})
```
Is there a way to create a mastodon api object here also using only my access_token (and url of course)?

Thanks!

**Comment by keponk** [comment]
hi @halcy looking at the tests from https://github.com/halcy/Mastodon.py/pull/132 trying to learn about this pytest.
If I understand correctly the basic change I made is in fact enough to allow the connection with Mastodon and allow previous users of this api to not notice the change. The tests however are written in a way that expect to have the client_id and so the tests need to be updated. Is that correct? 
I'll try to update the tests and will love some feedback on how they should be written to test for either access_token or client_id but, with one mandatory but not both. Thanks!

**Comment by keponk** [comment]
@halcy ok I think I'm getting the hang of it including the tests. The main issue is that I realize now some people might get their setup broken with this change. I was trying to figure out how allow access_token-only connection but now that I understand the tests better I see it's not possible.
If you'd like to allow connection using only access_token, that means the constructor needs to change. But with that change it means people currently calling the api using client_id as: `Mastodon ( '1a2b3c' )` will now get an error since making it optional would require them to switch to `Mastodon ( client_id='1a2b3c' )` at the very least.
How do you feel about that? I guess a note on the README and requesting a specific version from `pip` could solve the problem but I'd like to hear you opinion.

**Comment by halcy** [comment]
Oh, nah, it should still be possible to do this in a backward-compatible way. The constructor would have to be something like 

`_init__(self, client_id=None, access_token=None, client_secret=None,`

That could then be called either with client id as the first parameter, or like Mastodon(access_token="whatever"). Optional ("named") parameters in python can still be put in the order they were specified, so that would work!

The error handling would have to be changed (passing only an access token is not an error anymore), and if there is a test that checks for that it should probably be removed.

**Comment by halcy** [comment]
Closed by #134
